python/Ganga/Utility/execute.py

```diff
@@ -266,30 +267,40 @@ def execute(command,
             logger.error("stderr: %s" % stderr)
             raise RuntimeError("Missing update env after running command")
 
-    if not shell:
+    if not shell and not eval_includes:
         update_pkl_thread.join()
         if pkl_output_key in thread_output:
             return thread_output[pkl_output_key]
 
+    stdout_temp = None
     try:
+        # If output
         if stdout:
-            stdout = pickle.loads(stdout)
+            stdout_temp = pickle.loads(stdout)
     except pickle.UnpicklingError as err:
+        # If here, output likely unpickled
         if not shell:
-            logger.error("Execute Err: %s", err)
+            logger.debug("Execute Err: %s", err)
         else:
             logger.debug("Execute Err: %s", err)
-        local_ns = {}
+
+    if not stdout_temp:
+        local_ns = locals()
         if isinstance(eval_includes, str):
             try:
-                exec(eval_includes, {}, local_ns)
+                exec(eval_includes, local_ns)
```

Comment:
I doubt it makes a big difference but I'd have a mild preference to doing that simply because otherwise it looks like there's a specific reason to set the global ns as the local ns.